internal/migrations/migrator.go

```diff
@@ -160,6 +162,24 @@ func SetupMigrator(ctx context.Context, db Database, snapshotter Snapshotter, ac
 	}
 	migrator.lastChangeset = height
 
+	// migration status
+	if migrationParams.EndHeight == 0 {
```

Comment:
Ah nice this seems pretty good. I still think there is some simplification to be done between abci and migrator, but this seems good

Reply:
Actually no I don't think this works. This value will never be updated

Reply:
Yeah, pushing more changes, it's partially done rn